### PR TITLE
DOC: special: Expanded docs for Airy, elliptic, Bessel functions.

### PR DESCRIPTION
**Pull request by tpudlik**
The docs have been expanded by incorporating more information from the C and Fortran source files, especially concerning the algorithms used in the computation.

Any comments would be welcome: I'm hoping to eventually flesh out the documentation of all functions in `scipy.special` in similar style.

### CODE REVIEW

**Comment by rgommers** [comment]
Nice, this looks quite good @tpudlik!

One comment on the commits: it took me a while to realize that there weren't duplicate docstring additions, but that you had added them to ``add_newdocs.py`` and then ran ``generate_ufuncs.py``. It would be good to have the generated code added in separate commits.

**Comment by rgommers** [comment]
I've built the html docs - all looks good.

Now someone just needs to go through and review all the content, which will take a bit of time...

**Comment by tpudlik** [comment]
@rgommers, would you recommend doing a rebase to separate out the changes to `generate_ufuncs.py`, or should I just keep this in mind for the future?

**Comment by rgommers** [comment]
I'd say just keep in this mind for the future, a rebase is too much work for the gain now.

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -627,12 +672,40 @@ def add_newdoc(place, name, doc):
...
+
+    is used.
+
+    Accuracy:
+
+    ==========   ======     ========     =======     =======
+    arithmetic   domain     # trials      peak         rms
+    ==========   ======     ========     =======     =======
+       IEEE       0, 1       10000       2.1e-16     7.3e-17
+    ==========   ======     ========     =======     =======
```
Comment (ev-br) [inline scipy/special/add_newdocs.py]: I think this and similar tables should be omitted from the python-level docstrings: any benchmark results need to be reproducible, and these are not. At least I've no idea how to reproduce these numbers, do you? And unless I'm missing something obvious, I think the effort replicating these is better spent elsewhere (for instance: moving more information from C/Fortran comments to python docstrings).

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -627,12 +672,40 @@ def add_newdoc(place, name, doc):
     E : ndarray
         Value of the elliptic integral.
 
+    Notes
+    -----
+    Wrapper for the Cephes [1]_ routine `ellpe`.
+
+    For `m > 0` the computation uses the approximation,
+
+    .. math:: E(m) \\approx P(1-m) - (1-m) \\log(1-m) Q(1-m),
```
Comment (ev-br) [inline scipy/special/add_newdocs.py]: Minor comment for future: if you make the docstrings raw, you can use single backslashes too: , `r"""\frac{\epsilon}{\Theta^2}"""`. This way, TeX formulas look slightly more natural.

**Comment by ev-br** [comment]
This looks great, thank you for tackling it @tpudlik!

A couple of general comments:

* In the long run, it would be great to have explicit definitions for the functions, esp where there are multiple conventions in the literature (Condon-Schockley phase, specific definitions in the complex plane etc). Ideally, these would be supplemented by references to DLMF or A&S.

* Since this is math-heavy, I think we could use more TeX for typesetting. One thing I personally was finding very annoying is that we cannot use single dollar signs for short inline expressions, like `$ p-1 $` or phrases like "... where $n$ is integer". There are Sphinx extensions for the job, and Pauli made specific suggestions in https://github.com/numpy/numpydoc/pull/24.

None of these comments are meant as blocking this PR, all this can be done in subsequent PRs.

**Comment by tpudlik** [comment]
I was of two minds when it came to using TeX for typesetting.  On the one hand, if a variable appears in larger TeXed equations somewhere in the docstring, it should probably be TeXed throughout.  On the other hand, if a variable appears in monospaced font in the Parameters list, it would be good to render it in the same way in the rest of the docstring.  I was ultimately swayed by Numpy's [How to Document](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt), which advises that "LaTeX formatting should be kept to a minimum. Often it's possible to show equations as Python code or pseudo-code instead, which is much more readable in a terminal."

I'm happy edit the PR to use TeXed variants of variables exclusively if the docstring includes a TeX equation in these variables later.  Would this be better?  (I think it would be better for me personally, since I never read the SciPy docs in a terminal and can parse TeX anyway, but what is considered best practice?)

File: scipy/special/add_newdocs.py (hunk 1 above)
Comment (tpudlik) [inline scipy/special/add_newdocs.py]: This is fair: I'll remove these accuracy claims.

**Comment by ev-br** [comment]
Re TeX: the general recommendation is what you quote. W.r.t scipy.special, here's an appeal to authority: https://github.com/scipy/scipy/pull/2905#issuecomment-24861967

IMO it's a judgement call and it's very easy to overdo. There definitely is no need to make docstrings look like a source of a physics paper :-). Display math should be TeX. Small inline expressions could be either. In any case, there's no need to redo what you've done already.

**Comment by ev-br** [comment]
This is a clear improvement and there's no need to keep this in a limbo. Merging, thanks Ted! And keep them coming :-).

**Pull request merged**